src/utils/flattenMetadataQueryResponse.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @flow
+ * @file File for some simple dom utilities
+ * @author Box
+ */
+
+import omit from 'lodash/omit';
+import get from 'lodash/get';
+import type {
+    MetadataQueryResponse,
+    MetadataQeuryResponseEntry,
+    MetadataQeuryResponseEntryMetadata,
+} from '../common/types/metadataQueries';
+import { ITEM_TYPE_FILE } from '../common/constants';
+
+type flattenedMetadataResponseEntry = {
+    id?: string,
+    metadata?: {
+        data: Object,
+        id: string,
+        metadataTemplate: {
+            templateKey: string,
+            type: string,
+        },
+    },
+    name?: string,
+    size?: number,
+};
+
+const convertMetadata = (metadata: MetadataQeuryResponseEntryMetadata) => {
+    let md = {};
+
+    Object.keys(metadata).forEach(scope => {
+        Object.keys(metadata[scope]).forEach(templateKey => {
+            const nonconformingInstance = metadata[scope][templateKey];
+            const data = omit(nonconformingInstance, [
+                '$id',
+                '$parent',
+                '$type',
+                '$typeScope',
+                '$typeVersion',
+                '$version',
+            ]);
+
+            md = {
+                data,
+                id: nonconformingInstance.$id,
+                metadataTemplate: {
+                    type: 'metadata-template',
+                    templateKey,
+                },
+            };
+        });
+    });
+
+    return md;
+};
+
+const convertEntry = ({ item, metadata }: MetadataQeuryResponseEntry) => {
```

Comment:
Perhaps you could add the return flow types for these methods to clarify what the expected output will be